lock_api/src/mutex.rs

```diff
@@ -286,6 +289,45 @@ impl<R: RawMutex, T: ?Sized> Mutex<R, T> {
     pub fn data_ptr(&self) -> *mut T {
         self.data.get()
     }
+
+    /// # Safety
+    ///
+    /// The lock needs to be held for the behavior of this function to be defined.
+    #[cfg(feature = "arc_lock")]
+    #[inline]
+    unsafe fn guard_arc(this: &Arc<Self>) -> ArcMutexGuard<R, T> {
+        ArcMutexGuard {
+            mutex: this.clone(),
+            marker: PhantomData,
+        }
+    }
+
+    /// Acquires a lock through an `Arc`.
+    ///
+    /// This method is similar to the `lock` method; however, it requires the `Mutex` to be inside of an `Arc`
+    /// and the resulting mutex guard has no lifetime requirements.
+    #[cfg(feature = "arc_lock")]
+    #[inline]
+    pub fn lock_arc(this: &Arc<Self>) -> ArcMutexGuard<R, T> {
```

Comment:
```suggestion
    pub fn lock_arc(self: &Arc<Self>) -> ArcMutexGuard<R, T> {
```
You can use `self` here which allows `lock_arc` to be called with dot syntax.